include/soci/exchange-traits.h

```diff
@@ -43,66 +44,94 @@ struct exchange_traits
 };
 
 template <>
-struct exchange_traits<short>
+struct exchange_traits<int8_t>
 {
     typedef basic_type_tag type_family;
-    enum { x_type = x_short };
+    enum { x_type = x_int8 };
 };
 
 template <>
-struct exchange_traits<unsigned short> : exchange_traits<short>
+struct exchange_traits<uint8_t>
 {
+    typedef basic_type_tag type_family;
+    enum { x_type = x_uint8 };
 };
 
 template <>
-struct exchange_traits<int>
+struct exchange_traits<int16_t>
 {
     typedef basic_type_tag type_family;
-    enum { x_type = x_integer };
+    enum { x_type = x_int16 };
 };
 
 template <>
-struct exchange_traits<unsigned int> : exchange_traits<int>
+struct exchange_traits<uint16_t>
 {
+    typedef basic_type_tag type_family;
+    enum { x_type = x_uint16 };
 };
 
 template <>
-struct exchange_traits<char>
+struct exchange_traits<int32_t>
 {
     typedef basic_type_tag type_family;
-    enum { x_type = x_char };
+    enum { x_type = x_int32 };
 };
 
 template <>
-struct exchange_traits<long long>
+struct exchange_traits<uint32_t>
 {
     typedef basic_type_tag type_family;
-    enum { x_type = x_long_long };
+    enum { x_type = x_uint32 };
 };
 
 template <>
-struct exchange_traits<unsigned long long>
+struct exchange_traits<int64_t>
 {
     typedef basic_type_tag type_family;
-    enum { x_type = x_unsigned_long_long };
+    enum { x_type = x_int64 };
 };
 
-// long must be mapped either to x_integer or x_long_long:
-template<int long_size> struct long_traits_helper;
-template<> struct long_traits_helper<4> { enum { x_type = x_integer }; };
-template<> struct long_traits_helper<8> { enum { x_type = x_long_long }; };
-
 template <>
-struct exchange_traits<long int>
+struct exchange_traits<uint64_t>
 {
     typedef basic_type_tag type_family;
-    enum { x_type = long_traits_helper<sizeof(long int)>::x_type };
+    enum { x_type = x_uint64 };
+};
+
+#if defined(_WIN64)
```

Comment:
I don't quite like the idea of defining a "new" type for this. Instead of it, I added more descriptive defines. I _think_ that these can also handle the 32-bit case now.
Let me know what you think about this.